### PR TITLE
simplify code a bit

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

**Review by kinnison — CHANGES_REQUESTED** [review]
Hi Mathias, this is actually a clippy lint that we disagree with because it's making the shape of adding arguments different in the two places.

We have an issue open in clippy around this here - https://github.com/rust-lang/rust-clippy/issues/7071

I suppose in reality we should add an allow for that lint, just to keep things quieter, but we won't apply the refactor you suggest here.

**Comment by matthiaskrgr** [comment]
Well I got rid of the extra push entirely :upside_down_face:  but feel free to close if that's too convoluted. :sweat_smile:

File: tests/mock/clitools.rs [hunk 1]
```diff
@@ -637,17 +637,16 @@ where
     let mut vars: HashMap<String, String> = HashMap::default();
     self::env(config, &mut vars);
     vars.extend(env.iter().map(|(k, v)| (k.to_string(), v.to_string())));
-    let mut arg_strings: Vec<Box<str>> = Vec::new();
-    arg_strings.push(name.to_owned().into_boxed_str());
-    for arg in args {
-        arg_strings.push(
+    let arg_strings = Some(name.to_owned().into_boxed_str())
```
Comment (rbtcollins) [inline tests/mock/clitools.rs:640]: This is definitely not cleaner: iterating over a Some() to avoid a clippy lint... in test code.

**Review by rbtcollins — CHANGES_REQUESTED** [review]
Clarity is paramount in this test harness code - hypothetical performance advantages are irrelevant (at least for now), because test performance is dominated by the IO and layering we have, not in memory operations like this.

**Comment by matthiaskrgr** [comment]
Alright, closing ^^

**Pull request closed without merging**